test/types/zoneddatetime.jl

```diff
@@ -77,13 +77,13 @@ using Dates: Hour, Second, UTM, @dateformat_str
         utc_dt = DateTime(1916, 1, 31, 23)
 
         # Disambiguating parameters ignored when there is no ambiguity.
-        @test ZonedDateTime(local_dt, warsaw).zone.name == "CET"
-        @test ZonedDateTime(local_dt, warsaw, 0).zone.name == "CET"
-        @test ZonedDateTime(local_dt, warsaw, 1).zone.name == "CET"
-        @test ZonedDateTime(local_dt, warsaw, 2).zone.name == "CET"
-        @test ZonedDateTime(local_dt, warsaw, true).zone.name == "CET"
-        @test ZonedDateTime(local_dt, warsaw, false).zone.name == "CET"
-        @test ZonedDateTime(utc_dt, warsaw, from_utc=true).zone.name == "CET"
+        @test string(ZonedDateTime(local_dt, warsaw).zone.name) == "CET"
```

Comment:
Implementing promotion rules should avoid all these changes

Reply:
I don't think so. Happy to hear if i am wrong though.
Subtyping `AbstractString` would (but that is more work see https://github.com/JuliaTime/TimeZones.jl/pull/324#discussion_r611010337)
overloading equality would (but that can cause probelms since it would also require making hash agree and that is expensive to do)
AFIACT there is no promotion rule that gets invoked for `==` between `Strign` and a random type.